install/deploy_steps.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: MIT
+# Copyright (c) 2020 The Authors.
+
+# Authors: Hong Chang        <@Hong-Chang>
+#          Sherif Abdelwahab <@zasherif>
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:The above copyright
+# notice and this permission notice shall be included in all copies or
+# substantial portions of the Software.THE SOFTWARE IS PROVIDED "AS IS",
+# WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+# TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+# FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR
+# THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+source install/common.sh
+
+function deploy_steps:binaries_ready {
+    local is_mizar_production=${1:-0}
+    environment_adaptor:prepare_binary $is_mizar_production
+}
+
+function deploy_steps:environment_ready {
+    common:check_cluster_ready; local is_cluster_ready=$?
```

Comment:
I noticed that this function is checking the ready status of nodes.
In the deployment of a networking provider such as Mizar, Flannel, etc, we do not care about the status of nodes.

For example, when a cluster is brought up with no networking provider the nodes will always be in "NotReady" state. Once the networking provider is installed, the nodes will transition to  the "Ready" state. 

In the scenario described above, the deployment script will get stuck at this stage and Mizar will never not be installed.

Reply:
Changed to check cluster status